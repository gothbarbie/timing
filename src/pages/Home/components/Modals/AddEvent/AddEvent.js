import React from 'react'
import styled from 'styled-components'

import Button from '../../../../../generic/atoms/Button'

import LabeledInput from '../../../../../generic/molecules/LabeledInput'
import LabeledSelect from '../../../../../generic/molecules/LabeledSelect'
import LabeledTags from '../../../../../generic/molecules/LabeledTags'

const Wrapper = styled.section``

const TwoColumns = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 3rem;
  margin: 3rem 0;
`

const CategorySection = styled.div`
  display: grid;
  grid-template-columns: 6fr 1fr;
  grid-gap: 1rem;
  margin: 3rem 0;
  align-items: flex-end;
`

const AlignRight = styled.div`
  margin-top: 3rem;
  display: flex;
  justify-content: flex-end;
  width: 100%;
`

const AddEvent = () => (
  <Wrapper>
    <LabeledInput name="title" label="Title" />
    <TwoColumns>
      <LabeledInput name="start" label="Start" type="date" />
      <LabeledInput name="end" label="End" type="date" />
    </TwoColumns>
    <CategorySection>
      <LabeledSelect
        name="category"
        label="Category"
        options={[{ key: 'Test', value: 'TestValue' }]}
      />
      <Button primary>Edit</Button>
    </CategorySection>
    <TwoColumns>
      <LabeledInput name="add-tags" label="Add tags" />
      <LabeledTags
        tags={[{ blue: true, name: 'unit test', id: 1 }]}
        label="Used recently"
        onClick={id => console.log('added', id)}
      />
    </TwoColumns>
    <LabeledTags
      tags={[{ blue: false, name: 'story-125', id: 3 }]}
      label="Selected tags"
      onClick={id => console.log('deleted', id)}
    />
    <AlignRight>
      <Button primary>Create</Button>
    </AlignRight>
  </Wrapper>
)

export default AddEvent
