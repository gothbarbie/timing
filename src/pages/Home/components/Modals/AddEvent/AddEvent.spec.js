import React from 'react'
import { shallow } from 'enzyme'

import AddEvent from './AddEvent'

describe('pages/Home/components/Modals/AddEvent', () => {
  it('renders', () => {
    const component = shallow(<AddEvent />)

    expect(component).toMatchSnapshot()
  })
})
