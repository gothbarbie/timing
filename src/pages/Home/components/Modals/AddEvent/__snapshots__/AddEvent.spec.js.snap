// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Home/components/Modals/AddEvent renders 1`] = `
<styled.section>
  <LabeledInput
    label="Title"
    name="title"
  />
  <styled.div>
    <LabeledInput
      label="Start"
      name="start"
      type="date"
    />
    <LabeledInput
      label="End"
      name="end"
      type="date"
    />
  </styled.div>
  <styled.div>
    <LabeledSelect
      label="Category"
      name="category"
      options={
        Array [
          Object {
            "key": "Test",
            "value": "TestValue",
          },
        ]
      }
    />
    <Button
      primary={true}
    >
      Edit
    </Button>
  </styled.div>
  <styled.div>
    <LabeledInput
      label="Add tags"
      name="add-tags"
    />
    <LabeledTags
      label="Used recently"
      onClick={[Function]}
      tags={
        Array [
          Object {
            "blue": true,
            "id": 1,
            "name": "unit test",
          },
        ]
      }
    />
  </styled.div>
  <LabeledTags
    label="Selected tags"
    onClick={[Function]}
    tags={
      Array [
        Object {
          "blue": false,
          "id": 3,
          "name": "story-125",
        },
      ]
    }
  />
  <styled.div>
    <Button
      primary={true}
    >
      Create
    </Button>
  </styled.div>
</styled.section>
`;
